public/client.js

```diff
@@ -8,7 +8,7 @@ $(function () {
 		var list = [];
 		if (truckList) {
 			truckList.forEach(function (truck) {
-				list.push('<li><li><span class="delete_link" data-truck="' + truck._id + '">X</span><a href="/trucks/' + truck._id + '">' + truck.name + '</a></li>');
+				list.push('<li><li><span class="delete_link" data-truck="' + truck.name + '">X</span><a href="/trucks/' + truck.name + '">' + truck.name + '</a></li>');
```

Comment:
this should be _id so the value constructed matches the url for the delete handler